### PR TITLE
Allow custom GroupKFold object as split_type

### LINKED ISSUE #515: AttributeError: 'GroupKFold' object has no attribute 'groups'
When I set "split_type" as GroupKFold, then use fit() with setting "groups", I got the issue: _AttributeError: 'GroupKFold' object has no attribute 'groups'_ and

> ml.py:460, in evaluate_model_CV(config, estimator, X_train_all, y_train_all, budget, kf, task, eval_metric, best_val_loss, log_training_metric, fit_kwargs)
>     458     kf = kf.split(X_train_split, y_train_split)
>     459 elif isinstance(kf, GroupKFold):
> **_--> 460     groups = kf.groups_**
>     461     kf = kf.split(X_train_split, y_train_split, groups)
>     462     shuffle = False

So it seems the _groups_ value needs to be assigned in the **_evaluate_model_CV()_** function. Am I right?

### PR DESCRIPTION
**Pull request by sonichi**
close #515

Added additional checks for eval_method and split_type. When custom split_type is used, eval_method can't be "holdout".

### CODE REVIEW

**Comment by wuchihsu** [comment]
Is it needed to also add eval_method test in `test_object()`?

**Comment by sonichi** [comment]
> Is it needed to also add eval_method test in `test_object()`?

What about adding an assertion of `automl._state.eval_method=="cv"` after `automl.fit()`?

**Review by wuchihsu — APPROVED** [review]

File: test/automl/test_split.py [hunk 1]
```diff
@@ -80,6 +80,19 @@ def test_groups():
     automl_settings["eval_method"] = "holdout"
```
Comment (qingyun-wu) [inline test/automl/test_split.py:80]: Can eval_method be set to "auto"?

File: test/automl/test_split.py [hunk 2]
```diff
@@ -80,6 +80,19 @@ def test_groups():
...
     automl.fit(X, y, **automl_settings)
 
+    automl_settings["split_type"] = GroupKFold(n_splits=3)
+    try:
+        automl.fit(X, y, **automl_settings)
+        raise RuntimeError(
+            "GroupKFold object as split_type should fail when eval_method is holdout"
+        )
+    except AssertionError:
```
Comment (qingyun-wu) [inline test/automl/test_split.py:89]: Why catching AssertationError?

File: test/automl/test_split.py (hunk 1 above)
Comment (sonichi) [reply test/automl/test_split.py:80]: yes.

File: test/automl/test_split.py (hunk 2 above)
Comment (sonichi) [reply test/automl/test_split.py:89]: This is test code. We expect AssertionError to be raised in this case.

**Review by qingyun-wu — APPROVED** [review]

**Pull request merged**